src/modes/standard_race.cpp

```diff
@@ -98,36 +98,55 @@ void StandardRace::endRaceEarly()
     std::vector<int> active_players;
     // Required for debugging purposes
     beginSetKartPositions();
+    float worse_finish_time = 0.0f;
+
     for (unsigned int i = 1; i <= kart_amount; i++)
     {
         int kartid = m_position_index[i-1];
         AbstractKart* kart = m_karts[kartid];
+
         if (kart->hasFinishedRace())
         {
+            if (kart->getFinishTime() > worse_finish_time)
+                worse_finish_time = kart->getFinishTime();
+
             // Have to do this to keep endSetKartPosition happy
             setKartPosition(kartid, kart->getPosition());
-            continue;
         }
 
-        if (kart->getController()->isPlayerController())
+        else if (kart->getController()->isPlayerController())
         {
+            if (estimateFinishTimeForKart(kart) > worse_finish_time)
```

Comment:
Don't call estimateFinishTimeForKart twice, call it once and store the result in a variable, no use doing the calculation twice